client/components/domains/registrant-extra-info/uk-form.jsx

```diff
@@ -191,7 +198,10 @@ export class RegistrantExtraInfoUkForm extends React.PureComponent {
 			this.isTradingNameRequired( registrantType ) && 'tradingName',
 			this.isRegistrationNumberRequired( registrantType ) && 'registrationNumber',
 		] );
-		const relevantErrors = pick( get( validationErrors, 'extra.uk', {} ), relevantExtraFields );
+		const relevantErrors = pick(
+			get( contactDetailsValidationErrors, 'extra.uk', {} ),
```

Comment:
I'm not sure this is right, but it's hard to parse. Statements like this are why I try to avoid lodash when possible, especially now that JS itself has so many convenient tools that are easier to understand without reading docs.

If I turn this into pure JS, I think it becomes this:

```js
const relevantErrors = Object.keys( this.props.contactDetailsValidationErrors?.extra?.uk ?? {} )
  .reduce( ( errorsObject, errorKey ) => relevantExtraFields.includes( errorKey ) ? { ...errorsObject, [ errorKey ]: this.props.contactDetailsValidationErrors.extra.uk[ errorKey ] } : errorsObject, {} );
const isValid = Object.keys( relevantErrors ).length === 0;
```

And that sure is a lot of work to do just to set `isValid`. It seems to me that this would work just as well:

```js
const isValid = Object.keys( this.props.contactDetailsValidationErrors?.extra?.uk ?? {} )
  .every( errorKey => ! relevantExtraFields.includes( errorKey ) );
```

Furthermore, from what I know of how you're planning to use `contactDetailsValidationErrors`, is an object in `contactDetailsValidationErrors.extra.uk` really what you meant to do?

Reply:
For what it's worth, that line only appears in this diff because of the `validationErrors -> contactDetailsValidationErrors` renaming. I am totally ok with getting rid of some lodash where we can though. :)

I think what's happening here is that which fields are required in this form depends on the selected registrant type, and the component is doing some logic to validate this on the frontend. I think your rewrite is correct.

As for your last question, if I understand correctly, the answer is yes; if there are validation errors in these fields, the object returned by the endpoint has an object at `extra.uk` with a property for each errored field.

Reply:
Ok! If the data is correct, that's fine. Now that I re-read my comment it sounds a little like I was saying "how dare you use lodash" but I really meant "since this existing code uses lodash, it's hard for me to parse the effects of your change". I'll leave the decision about what to do here to you. 😄

Reply:
Haha- no worries. I also find lodash to be confusing, having come to JS after the era in which it was necessary. A lot of times I end up leaving existing code alone even when it probably could be made more clear because I'm not confident I won't break something.